javascript/reflex.js

```diff
@@ -0,0 +1,34 @@
+import Debug from './debug'
+
+export default class Reflex {
+  constructor (data, controller) {
+    this.data = data.valueOf()
+    this.controller = controller
+    this.element = data.reflexElement
+    this.reflexId = data.reflexId
```

Comment:
I've been thinking about this a lot (especially while I was working on the PR). The origin of `reflexId` was when I wanted to attach the then newly created reflex id to the Promise returned by `stimulate`.

1. It might come down to "this will work on the client but not the server" (or vice versa) TBD
2. What I might do on the client is create a getter for `reflexId` that emits a `console.warn` deprecation notice that devs should switch over to `id` on the Promise; I'm not sure it exists elsewhere in the public API for 3.4.x
3. Thinking about attaching variables to that Promise... instead of switching to `id` here, it probably makes a lot more sense to return a `reflex` property instead, so they could use `this.stimulate(...).reflex.id` and have access to the whole object instance.

I'll kick the tires and see what gotchas fall out.

Reply:
I believe that I have this working well. I was originally going to attempt to keep `reflex_id` in the request payload, but I suspect that this will just cause more confusion in the long run. @joshleblanc will likely have to update VCR to account for the name change.

The Promise now supports `id`, `reflex` and `reflexId` queries. If you query `reflexId`, it returns the `id` and offers a deprecation warning. `reflex` is just a pointer to the Reflex created by the `stimulate` call.

I also modified the `stimulate` options processor to accept both `id` and `reflexId`; if they pass a `reflexId` it offers a deprecation warning and silently renames the option to `id`.

```js
    const x = this.stimulate('example#ping', { reflexId: 666 })
    console.log(x.id)
    console.log(x.reflexId)
    console.log(x.reflex)
    x.then(payload => {
      console.log('itza payload', payload)
    })
```
<img width="727" alt="chrome_9SaatZxKtO" src="https://user-images.githubusercontent.com/38150464/183568022-e8e0f046-baa0-4055-b494-27abba78a090.png">